### PR TITLE
Dynamic, resizable camera components

### PR DESCRIPTION
**Pull request by starphys**
Cameras currently use statically or publicly hosted images, but we will pass in urls and ratios for the actual rover cameras in the future. They can be resized by dragging the boundaries of the component, and they will respond to either a constrained height or width appropriately. Right clicking opens up a context menu to switch between images.

Big shoutout to @NathanaelG2099 for getting this started here: https://github.com/NathanaelG2099/psychic-adventure

### CODE REVIEW

**Comment by starphys** [comment]
@naterpotatoers can you peep the CSS on this and let me know if you have any tips? Just getting this to work at all was a slog, and I wouldn't be surprised if there are much better ways to do it.

File: src/panes/CameraPane.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      sx={{
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center'
+      }}
+    >
+      <img
+        style={{ ...imageStyle, maxHeight: '100%', maxWidth: '100%' }}
```
Comment (naterpotatoers) [inline src/panes/CameraPane.jsx]: maybe object-fit: contain could work instead of maxHeight/Width?

File: src/panes/CameraPane.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  const imageStyle = calculateImageDimensions()
+  return (
+    <Box
+      ref={containerRef}
+      sx={{
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'column',
```
Comment (naterpotatoers) [inline src/panes/CameraPane.jsx:44]: Grid does this by default but not exactly the same in terms of responsiveness. Might be slight improvement 🤷 
```suggestion
        display: 'grid',
```

**Review by naterpotatoers — APPROVED** [review]
Didn't clone and test out recommendations but just looking things over this morning I don't see any glaring red flags. Seems like you guys are just building a pretty complex interface. I'm sure there is a way to optimize better but for now this looks okay to me

File: src/panes/CameraPane.jsx (hunk 1 above)
Comment (starphys) [reply src/panes/CameraPane.jsx]: This was one of the first things I tried, but couldn't get it to work correctly. I may just have been using it wrong, or possibly had incorrect css on the parent div. ~~For now I'll merge as is, and leave it as an exercise to find a simpler way to do this.~~

Edit: it works the same and looks a bit cleaner with your suggestion, so I'm taking it.

File: src/panes/CameraPane.jsx (hunk 2 above)
Comment (starphys) [reply src/panes/CameraPane.jsx:44]: For whatever reason this causes panes to overlap one another instead of resizing.

**Pull request merged**